doc/sphinx/proof-engine/ltac.rst

```diff
@@ -469,45 +500,117 @@ We can check if a tactic made progress with:
+
+The :tacn:`+<+ (backtracking branching)>`, :tacn:`multimatch` and :tacn:`multimatch goal` tactics
+provide more complex capability.  Rather than applying a single successful
+tactic, these tactics generate a series of successful tactic alternatives that are tried sequentially
+when subsequent tactics outside these constructs fail.  Please examine the :tacn:`multimatch` example
+:ref:`here<match_vs_multimatch_ex>`.
+
+These constructs are evaluated using :gdef:`backtracking`.  Each  creates a
+:gdef:`backtracking point`.  When a subsequent tactic fails, evaluation continues from the nearest
+prior backtracking point with the next successful alternative and repeats the tactics after
+the backtracking point.  When a backtracking point has
+no more successful alternatives, evaluation continues from the next prior backtracking point.
+If there are no more prior backtracking points, the overall tactic fails.
+
+Thus, backtracking tactics can have multiple successes.  Non-backtracking constructs that appear
+after a backtracking point are reprocessed after backtracking, as in the example
+:ref:`here<match_vs_multimatch_ex>`, in which
+the :tacn:`;<ltac-seq>` construct is reprocessed after backtracking.  When a backtracking construct is within
+a non-backtracking construct, the latter uses the :gdef:`first success`.  Backtracking to
+a point within a non-backtracking construct won't change the branch that was selected by the
+non-backtracking construct.
+
+The :tacn:`once` tactic stops further backtracking to backtracking points within that tactic.
+
+eval vs execution (also out of order)
+
+Branching with backtracking: +
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can branch with backtracking with the following structure:
+
+.. tacn:: @ltac_expr1 + {| @ltac_expr2 | @binder_tactic }
    :name: + (backtracking branching)
 
-   :n:`@ltac_expr__1` and :n:`@ltac_expr__2` are evaluated to :n:`v__1` and
+   :n:`@ltac_expr1` and :n:`@ltac_expr2` are evaluated to :n:`v__1` and
    :n:`v__2`  respectively, which must be tactic values. The tactic value :n:`v__1` is applied to
    each focused goal independently and if it fails or a later tactic fails, then
    the proof backtracks to the current goal and :n:`v__2` is applied.
+   (To prevent evaluation of further alternatives, use :tacn:`first` instead.)
 
    Tactics can be seen as having several successes. When a tactic fails it
    asks for more successes of the prior tactics.
-   :n:`@ltac_expr__1 + @ltac_expr__2` has all the successes of :n:`v__1` followed by all the
+   :n:`@ltac_expr1 + @ltac_expr2` has all the successes of :n:`v__1` followed by all the
    successes of :n:`v__2`. Algebraically,
-   :n:`(@ltac_expr__1 + @ltac_expr__2); @ltac_expr__3 = (@ltac_expr__1; @ltac_expr__3) + (@ltac_expr__2; @ltac_expr__3)`.
+   :n:`(@ltac_expr1 + @ltac_expr2); @ltac_expr3 = (@ltac_expr1; @ltac_expr__3) + (@ltac_expr2; @ltac_expr3)`.
 
    Branching is left-associative.
```

Comment:
Do we want the example of how it's not fully associative with failure levels?

Reply:
Sure, hopefully short and easy to understand.  BTW, is there anywhere where right vs left associativity matters to users?  I've not run across a place but I haven't taken a thorough look.

Reply:
Again, this PR discussion is getting too big to navigate.  I had to download all the comments via the API and search though them on the command line to find the example.  It's at https://github.com/coq/coq/pull/12103#discussion_r432900017 : 
```coq
Goal True.
  Fail (fail 2 + idtac) + idtac. (* Tactic failure. *)
  Fail fail 2 + (idtac + idtac). (* Tactic failure (level 1). *)
```
As far as I can tell, this is the only place where associativity of `+` shows up in a user-visible way, and I've literally never encountered this case in real code.  (It's very, very rare to mix failure levels with `+`, because they're created to solve overlapping but distinct problems.)  (Left vs right associativity matters a lot more for other constructs, such as `; [ ... ]` in Ltac and `::` in Gallina (list cons).)

Reply:
> Again, this PR discussion is getting too big to navigate.

Yup.  I think we're almost done, though.  To view everything, I search in the browser for "hidden " and expand all those, then search for "resolve ".  Or I expand all the notification emails in Gmail, do a text search in the browser for what I want and click on the GitHub link below the relevant message to get to the exact spot.